src/main/java/org/apache/maven/plugins/dependency/analyze/AbstractAnalyzeMojo.java

```diff
@@ -49,7 +49,7 @@
 
 /**
  * Analyzes the dependencies of this project and determines which are: used and declared; used and undeclared; unused
- * and declared. Also determines which dependencies have a broader scope than needed.
+ * and declared; non-test scoped test-only dependencies.
```

Comment:
non-test scoped test-only dependencies --> compile scoped but only used in tests

Reply:
It also works if a test only dependency is scoped with runtime, provided, or compile so how about; runtime, provided, or compile scoped but only used in tests?

Reply:
That sounds like an issue. We probably don't want to suggest changing runtime scope to test scope since the analyzer can't detect how a runtime dependency is used.

Reply:
Ok, the test fails if it's scoped with runtime but used in non test classes but it seems like it may be failing for the wrong reason. Gonna look into this more. If it needs to ignore runtime scopes that shouldn't be too complicated but will need to be changed in the actual dependency-analyzer project.